rust/cookiecutter-aws-sam-hello-rust-dynamodb/{{ cookiecutter.project_slug }}/src/main.rs

```diff
@@ -0,0 +1,131 @@
+use aws_sdk_dynamodb::{model::AttributeValue, Client};
+use lambda_http::{
+    ext::RequestExt,
+    handler,
+    lambda_runtime::{self, Context, Error},
+    Body, IntoResponse, Request, Response,
+};
+use std::env;
+
+#[tokio::main]
+async fn main() -> Result<(), Error> {
+    let config = aws_config::load_from_env().await;
+    let table_name = env::var("TABLE_NAME").expect("TABLE_NAME must be set");
+    let dynamodb_client = Client::new(&config);
+
+    lambda_runtime::run(handler(|request: Request, context: Context| {
+        put_item(&dynamodb_client, &table_name, request, context)
+    }))
+    .await?;
+
+    Ok(())
+}
+
+async fn put_item<C>(
+    client: &Client<C>,
+    table_name: &str,
+    request: Request,
+    _context: Context,
+) -> Result<impl IntoResponse, Error>
+where
+    C: aws_smithy_client::bounds::SmithyConnector,
+{
+    // Extract path parameter from request
+    let path_parameters = request.path_parameters();
+    let id = match path_parameters.get("id") {
+        Some(id) => id,
+        None => return Ok(Response::builder().status(400).body("id is required")?),
+    };
+
+    // Extract body from request
+    let body = match request.body() {
+        Body::Empty => "".to_string(),
+        Body::Text(body) => body.clone(),
+        Body::Binary(body) => String::from_utf8_lossy(body).to_string(),
+    };
+
+    // Put the item in the DynamoDB table
+    let res = client
+        .put_item()
+        .table_name(table_name)
+        .item("id", AttributeValue::S(id.to_string()))
+        .item("payload", AttributeValue::S(body))
+        .send()
+        .await;
+
+    // Return a response to the end-user
+    match res {
+        Ok(_) => Ok(Response::builder().status(200).body("item saved")?),
+        Err(_) => Ok(Response::builder().status(500).body("internal error")?),
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
Is it typical in Rust to mix test code with regular code? From a Rust newbie's perspective, this seems very verbose (larger than the actual code) and hard to distinguish. Would it make sense to separate them?

Reply:
Yes, that's a pretty typical way to organize a Rust project: you usually keep the unit tests close to the code. I can add some documentation to make it clearer.